odoo/addons/base/tests/test_ir_actions.py

```diff
@@ -319,17 +319,16 @@ class TestCustomFields(common.TransactionCase):
 
     def setUp(self):
         # check that the registry is properly reset
-        registry = odoo.registry()
-        fnames = set(registry[self.MODEL]._fields)
+        fnames = set(self.registry[self.MODEL]._fields)
+
         @self.addCleanup
         def check_registry():
-            assert set(registry[self.MODEL]._fields) == fnames
+            assert set(self.registry[self.MODEL]._fields) == fnames
 
-        super(TestCustomFields, self).setUp()
+        self.addCleanup(self.registry.reset_changes)
+        self.addCleanup(self.registry.clear_caches)
 
-        # use a test cursor instead of a real cursor
-        self.registry.enter_test_mode(self.cr)
-        self.addCleanup(self.registry.leave_test_mode)
+        super(TestCustomFields, self).setUp()
```

Comment:
Github (wrongly) detects a diff here, let's take the opportunity to change py2's super to py3